src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1569,6 +1580,71 @@ protected function hasChanges($changes, $attributes = null)
         return false;
     }
 
+    /**
+     * Determine if the attribute is transitioning to a given value.
+     *
+     * @param  string  $attribute
+     * @param  string|array  $to
+     * @param  string|array  $from
+     * @return bool
+     */
+    public function changingTo($attribute, $to, $from = null)
+    {
+        if (! $this->isDirty($attribute)) {
+            return false;
+        }
+
+        $oldValue = is_null($this->original[$attribute]) ? null : (string) $this->original[$attribute];
+        $newValue = is_null($this->attributes[$attribute]) ? null : (string) $this->attributes[$attribute];
+
+        if (func_num_args() == 3 && ! in_array($oldValue, is_array($from) ? $from : [$from])) {
+            return false;
+        }
+
+        return in_array($newValue, is_array($to) ? $to : [$to]);
+    }
+
+    /**
+     * Determine if the attribute has transitioned to a given value.
+     *
+     * @param  string  $attribute
+     * @param  string|array  $to
+     * @param  string|array  $from
+     * @return bool
+     */
+    public function changedTo($attribute, $to, $from = null)
+    {
+        if (! array_key_exists($attribute, $this->previousAttributes) ||
+            ! array_key_exists($attribute, $this->changes)) {
+            return false;
+        }
+
+        $oldValue = is_null($this->previousAttributes[$attribute]) ? null : (string) $this->previousAttributes[$attribute];
+        $newValue = is_null($this->changes[$attribute]) ? null : (string) $this->changes[$attribute];
+
+        if (func_num_args() == 3 && ! in_array($oldValue, is_array($from) ? $from : [$from])) {
+            return false;
+        }
+
+        return in_array($newValue, is_array($to) ? $to : [$to]);
+    }
+
+    /**
+     * Get an instance of the model in its previous state.
+     *
+     * @return static
+     */
+    public function wormhole()
+    {
+        if (empty($this->previousAttributes)) {
+            throw new LogicException("Unable to wormhole. Model has not been saved.");
+        }
+
+        return tap(new static, function ($instance) {
+            $instance->setRawAttributes($this->previousAttributes, true);
+        });
```

Comment:
Doesn't `setRawAttributes` return `$this`?

```suggestion
        return (new static)->setRawAttributes($this->previousAttributes, true);
```